internal/security/credentials.go

```diff
@@ -153,5 +164,20 @@ func (s *SecretProvider) updateSecretsCache(path string, secrets map[string]stri
 // path specifies the type or location of the secrets to store
 // secrets map specifies the "key": "value" pairs of secrets to store
 func (s *SecretProvider) StoreSecrets(path string, secrets map[string]string) error {
-	return nil // TODO: returning nil until interface is implemented
+
+	if !s.isSecurityEnabled() {
+		return errors.New("Storing secrets is not supported when running insecure mode")
+	}
+
+	err := s.SecretClient.StoreSecrets(path, secrets)
```

Comment:
Should we validate input first or will StoreSecrets do any input validation

Reply:
Added validation to the web handler